common/src/main/java/com/wynntils/gui/screens/PartyManagementScreen.java

```diff
@@ -0,0 +1,479 @@
+
+            // head rendering
+            RenderUtils.drawTexturedRect(
+                    poseStack,
+                    McUtils.mc().getConnection().getPlayerInfo(playerName).getSkinLocation(),
+                    this.width / 2 - xStart + 4,
+                    this.height / 2 - 125 + i * 20 - 10,
+                    8,
+                    16,
+                    16,
+                    8,
+                    8,
+                    8,
+                    8,
+                    64,
+                    64);
+
+            // name rendering
+            CustomColor color = playerName.equals(Models.Party.getPartyLeader()) ? CommonColors.YELLOW : (Models.Friends.getFriends().contains(playerName) ? CommonColors.GREEN : CommonColors.WHITE);
+            String prefix = offlineMembers.contains(playerName) ? "§m" : (playerName.equals(McUtils.player().getName().getString()) ? "§l" : "");
+
+            String formattedPlayerName = prefix + playerName;
+
+            fr.renderText(
+                    poseStack,
+                    formattedPlayerName,
+                    this.width / 2 - xStart + 40,
+                    this.height / 2 - 125 + i * 20,
+                    color,
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Middle,
+                    TextShadow.NORMAL);
+
+            if (!McUtils.player().getName().getString().equals(Models.Party.getPartyLeader())) continue; // only leader can promote/kick
+
+            if (playerName.equals(Models.Party.getPartyLeader())) {
+                kickButtons.add(new Button.Builder(
+                        Component.translatable("screens.wynntils.partyManagementGui.disband"),
+                        (button) -> disbandParty())
+                        .pos(this.width / 2 - xStart + 296, this.height / 2 - 125 + i * 20 - 10)
+                        .size(50, 20)
+                        .build());
+                continue;
+            }
+
+            // Promote button
+            promoteButtons.add(new Button.Builder(
+                    Component.translatable("screens.wynntils.partyManagementGui.promote"),
+                    (button) -> promoteToLeader(playerName))
+                    .pos(this.width / 2 - xStart + 244, this.height / 2 - 125 + i * 20 - 10)
+                    .size(50, 20)
+                    .build());
+
+            // Kick button
+            kickButtons.add(new Button.Builder(
+                    Component.translatable("screens.wynntils.partyManagementGui.kick"),
+                    (button) -> kickFromParty(playerName))
+                    .pos(this.width / 2 - xStart + 296, this.height / 2 - 125 + i * 20 - 10)
+                    .size(50, 20)
+                    .build());
+        }
+        this.renderables.addAll(promoteButtons);
+        this.renderables.addAll(kickButtons);
+        this.children.addAll(promoteButtons);
+        this.children.addAll(kickButtons);
```

Comment:
This is really bad, we can't do such things in render time.. Can't you do this in init, or have some kind of update event that updates periodically?

Reply:
Also, use this.addRenderableWidget(). For removing, well, you shouldn't really remove widgets without re-initing...

Reply:
Okaay, I see now. Make these a widget. And make that widget render the promote, kick, player head, etc. It results in nicer code.

Reply:
(I know this is WIP, but doing things the proper way by default is less work in the future.)